x-pack/plugins/security/server/session_management/session_index.ts

```diff
@@ -317,69 +317,75 @@ export class SessionIndex {
 
```

Comment:
__note:__ something weird happened to the position of the code comments in this function (moved to a wrong line, and also unnecessary new lines appeared). GitHub doesn't allow me to insert suggestions as changes include deleted lines, would you mind fixing\reverting some of these?

By the way, actually here we intentionally and carefully tried to wrap all the places where we expect errors in `try/catch` blocks to have distinct error log entries (excluding developer mistakes where we wanted everything to crash immediately during development :slightly_smiling_face: ). But I understand that it's close to impossible to make ESLint rule aware of something like this without ignoring the rule completely, so "global" try-catch-just-in-case is fine by me.

Reply:
Yeah, I definitely noticed a solid attempt to capture all the errors, and the only thing that runs outside of the `try/catch` is the logging methods, but utilizing unhandled rejections to take the server down I think is probably a pattern we want to avoid.

I've fixed the formatting, that's a symptom of Babel parsing the code and reformatting it from raw babel transformations I think.

Reply:
Also, it feels like it would be a lot better to format this specific code as:

```ts
return (this.indexInitialization = (async () => {
  // Check if required index template exists.
  let indexTemplateExists = false;
  try {
    indexTemplateExists = (
      await this.options.elasticsearchClient.indices.existsTemplate({
        name: sessionIndexTemplateName,
      })
    ).body;
  } catch (err) {
    this.options.logger.error(
      `Failed to check if session index template exists: ${err.message}`
    );
    throw err;
  }

  //...

})());
```

Reply:
> I've fixed the formatting, that's a symptom of Babel parsing the code and reformatting it from raw babel transformations I think.

Thanks!

> Or just break this function out into a private method and call it to get the promise.

Yeah, that's probably what we'll do eventually.